Odd failures in the builds:

```
[2026-05-31T11:39:47.811Z] [INFO] ------------------------------------------------------------------------
[2026-05-31T11:39:47.811Z] [INFO] BUILD FAILURE
[2026-05-31T11:39:47.811Z] [INFO] ------------------------------------------------------------------------
[2026-05-31T11:39:47.811Z] [INFO] Total time:  01:21 min
[2026-05-31T11:39:47.811Z] [INFO] Finished at: 2026-05-31T11:39:47Z
[2026-05-31T11:39:47.811Z] [INFO] ------------------------------------------------------------------------
[2026-05-31T11:39:48.069Z] [ERROR] Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:5.0.3:compare-version-with-baselines (compare-attached-artifacts-with-release) on project org.eclipse.jdt.core.compiler.batch: Baseline and reactor have the same fully qualified version, but different content
[2026-05-31T11:39:48.069Z] [ERROR] different
[2026-05-31T11:39:48.069Z] [ERROR]    META-INF/ECLIPSE_.RSA: not present in baseline
[2026-05-31T11:39:48.069Z] [ERROR]    META-INF/ECLIPSE_.SF: not present in baseline
[2026-05-31T11:39:48.069Z] [ERROR] -> [Help 1]
[2026-05-31T11:39:48.069Z] org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:5.0.3:compare-version-with-baselines (compare-attached-artifacts-with-release) on project org.eclipse.jdt.core.compiler.batch: Baseline and reactor have the same fully qualified version, but different content
[2026-05-31T11:39:48.069Z] different
[2026-05-31T11:39:48.069Z]    META-INF/ECLIPSE_.RSA: not present in baseline
[2026-05-31T11:39:48.069Z]    META-INF/ECLIPSE_.SF: not present in baseline
[2026-05-31T11:39:48.069Z] 
[2026-05-31T11:39:48.069Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
[2026-05-31T11:39:48.069Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
[2026-05-31T11:39:48.069Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
[2026-05-31T11:39:48.069Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
[2026-05-31T11:39:48.069Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
[2026-05-31T11:39:48.069Z]     at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
[2026-05-31T11:39:48.069Z]     at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
[2026-05-31T11:39:48.070Z]     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
[2026-05-31T11:39:48.070Z]     at java.lang.reflect.Method.invoke (Method.java:565)
[2026-05-31T11:39:48.070Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
[2026-05-31T11:39:48.070Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
[2026-05-31T11:39:48.070Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
[2026-05-31T11:39:48.070Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
[2026-05-31T11:39:48.070Z] Caused by: org.apache.maven.plugin.MojoFailureException: Baseline and reactor have the same fully qualified version, but different content
[2026-05-31T11:39:48.070Z] different
[2026-05-31T11:39:48.070Z]    META-INF/ECLIPSE_.RSA: not present in baseline
[2026-05-31T11:39:48.070Z]    META-INF/ECLIPSE_.SF: not present in baseline
[2026-05-31T11:39:48.070Z] 
[2026-05-31T11:39:48.070Z]     at org.eclipse.tycho.plugins.p2.extras.CompareWithBaselineMojo.execute (CompareWithBaselineMojo.java:229)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
[2026-05-31T11:39:48.070Z]     at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
[2026-05-31T11:39:48.070Z]     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
[2026-05-31T11:39:48.070Z]     at java.lang.reflect.Method.invoke (Method.java:565)
[2026-05-31T11:39:48.070Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
[2026-05-31T11:39:48.070Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
[2026-05-31T11:39:48.070Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
[2026-05-31T11:39:48.070Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
[2026-05-31T11:39:48.070Z] Caused by: org.apache.maven.plugin.MojoFailureException: Baseline and reactor have the same fully qualified version, but different content
[2026-05-31T11:39:48.070Z] different
[2026-05-31T11:39:48.070Z]    META-INF/ECLIPSE_.RSA: not present in baseline
[2026-05-31T11:39:48.070Z]    META-INF/ECLIPSE_.SF: not present in baseline
[2026-05-31T11:39:48.070Z] 
[2026-05-31T11:39:48.070Z]     at org.eclipse.tycho.plugins.p2.extras.CompareWithBaselineMojo.execute (CompareWithBaselineMojo.java:210)
[2026-05-31T11:39:48.071Z]     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
[2026-05-31T11:39:48.071Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
[2026-05-31T11:39:48.071Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
[2026-05-31T11:39:48.071Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
[2026-05-31T11:39:48.071Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
[2026-05-31T11:39:48.071Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
[2026-05-31T11:39:48.071Z]     at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
[2026-05-31T11:39:48.071Z]     at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
[2026-05-31T11:39:48.071Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
[2026-05-31T11:39:48.071Z]     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
[2026-05-31T11:39:48.071Z]     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
[2026-05-31T11:39:48.071Z]     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
[2026-05-31T11:39:48.071Z]     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
[2026-05-31T11:39:48.071Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
[2026-05-31T11:39:48.071Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
[2026-05-31T11:39:48.071Z]     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
[2026-05-31T11:39:48.071Z]     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
[2026-05-31T11:39:48.071Z]     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
[2026-05-31T11:39:48.071Z]     at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
[2026-05-31T11:39:48.071Z]     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
[2026-05-31T11:39:48.071Z]     at java.lang.reflect.Method.invoke (Method.java:565)
[2026-05-31T11:39:48.071Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
[2026-05-31T11:39:48.071Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
[2026-05-31T11:39:48.071Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
[2026-05-31T11:39:48.071Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
```

--- a/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/CopyMoveElementsTests.java
+++ b/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/CopyMoveElementsTests.java
@@ -403,6 +403,7 @@ public void testCopyFieldsMultiStatus() throws CoreException {
 	}
 	dests[1] = fieldsSource[0]; //invalid destination
 	dests[2]=  fieldsSource[0];
+	waitForAutoRefresh();
 	DeltaListener listener = new DeltaListener();
 	try {
 		startDeltas(listener);

--- a/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/CopyMoveTests.java
+++ b/org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/CopyMoveTests.java
@@ -59,6 +59,8 @@ public void copyNegative(IJavaElement[] elements, IJavaElement[] destinations, I
  * encountered are thrown.
  */
 public IJavaElement copyPositive(IJavaElement element, IJavaElement container, IJavaElement sibling, String rename, boolean force) throws CoreException {
+	// make sure auto-refresh is not interfering with the test, see e.g.: https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2552
+	waitForAutoRefresh();
 	// if forcing, ensure that a name collision exists
 	if (force) {
 		IJavaElement collision = generateHandle(element, rename, container);

